cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceUtils.java

```diff
@@ -144,4 +146,46 @@ public static void updateKafkaAutoRebalanceStatus(KafkaStatus kafkaStatus, Kafka
             }
         }
     }
+
+    /**
+     * Updates the {@code KafkaRebalanceStatus} with a new Warning condition based
+     * on the provided {@link Throwable}. If a Warning condition with the same reason
+     * and message already exists, no update is performed.
+     *
+     * @param status    The {@link KafkaRebalanceStatus} object whose conditions will be updated.
+     * @param exception The {@link Throwable} containing the reason and message for the Warning condition.
+     */
+    public static void addWarningCondition(KafkaRebalanceStatus status, Throwable exception) {
+        List<Condition> conditions = status.getConditions() != null ? new ArrayList<>(status.getConditions()) : new ArrayList<>();
+
+        String causeDescription = exception.getCause() != null
+                ? exception.getCause().toString()
+                : "No cause available";
+        Condition newCondition = StatusUtils.buildWarningCondition(causeDescription, exception.getMessage());
+        Condition oldCondition = conditions.stream()
+                .filter(c -> "Warning".equals(c.getType()))
+                .findFirst()
+                .orElse(null);
+
+        if (oldCondition != null) {
+            // If existing Warning condition has same reason & message, do nothing
+            if (Objects.equals(newCondition.getReason(), oldCondition.getReason()) &&
+                Objects.equals(newCondition.getMessage(), oldCondition.getMessage())) {
+                return;
+            }
+            // Otherwise, remove existing Warning condition
+            conditions.remove(oldCondition);
+        }
+
+        // Add new Warning condition
+        conditions.add(newCondition);
+        status.setConditions(conditions);
+    }
+
+    /* test */ static Condition getWarningCondition(KafkaRebalanceStatus status) {
```

Comment:
What is this method used for? You don't seem to use it above?

Reply:
This was supposed to be used by the method above it, just fixed.